lib/shopify_app/sessions_controller.rb

```diff
@@ -16,6 +16,7 @@ def callback
         sess = ShopifyAPI::Session.new(shop_name, response['credentials']['token'])
         session[:shopify] = ShopifyApp::SessionRepository.store(sess)
         session[:shopify_domain] = shop_name
+        session[:return_to] = nil
```

Comment:
Its complicated stuff because there are so many possible states. I don't think what you've got there is correct because we only want to clear it in the callback endpoint. We want it to carry through the other ones